File: Design-Patterns-Using-Java/Basics OOPS/CompositionRelationship.java
import java.util.ArrayList;
import java.util.List;

class Tyre {
    private String type;

    public Tyre(String type) {
        this.type = type;
    }

    public String getType() {
        return type;
    }
}

class Car {
    private String model;
    private List<Tyre> tyres;

    public Car(String model) {
        this.model = model;
        this.tyres = new ArrayList<>();
        for (int i = 0; i < 4; i++) {
            this.tyres.add(new Tyre("MRF"));
        }
    }

    public void display() {
        System.out.println("Car: " + model + ", Tyre: " + tyres.get(0).getType());
    }
}

public class CompositionRelationship {
    public static void main(String[] args) {
        Car car = new Car("BMW");
        car.display();
    }
}

/*
Car: BMW, Tyre: MRF
 */

File: Design-Patterns-Using-Swift/Basics OOPS/CompositionRelationship.swift
class Tyre {
    private var type: String

    init(type: String) {
        self.type = type
    }

    func getType() -> String {
        return type
    }
}

class Car {
    private var model: String
    private var tyres: [Tyre]

    init(model: String) {
        self.model = model
        self.tyres = []
        for _ in 0..<4 {
            self.tyres.append(Tyre(type: "MRF"))
        }
    }

    func display() {
        print("Car: \(model), Tyre: \(tyres[0].getType())")
    }
}

let car = Car(model: "BMW")
car.display()
